modules/constructor/constructor-basic/constructor-basic.services.js

```diff
@@ -0,0 +1,99 @@
+const ConstructorBasic = require('./constructor-basic.model');
+const uploadService = require('../../upload/upload.service');
+const {
+  BASIC_NOT_FOUND,
+  BASIC_ALREADY_EXIST,
+  IMAGE_NOT_PROVIDED,
+} = require('../../../error-messages/constructor-basic-messages');
+const {calculatePrice} = require('../../../utils/calculate-price');
+
+class ConstructorBasicService {
+  async getAllConstructorBasics({ skip, limit }) {
+    const items = await ConstructorBasic.find().populate(
+      'material',
+    )
+      .skip(skip)
+      .limit(limit);
+
+    const count = await ConstructorBasic.find().countDocuments();
+    return {
+      items,
+      count,
+    };
+  }
+
+  async getConstructorBasicById(id) {
+    const foundBasic = await ConstructorBasic.findById(id).populate(
+      'material',
+    );
+    if (foundBasic) {
+      return foundBasic;
+    }
+    throw new Error(BASIC_NOT_FOUND);
+  }
+
+  async addConstructorBasic(data, upload) {
+    if (await this.checkConstructorBasicExist(data)) {
+      throw new Error(BASIC_ALREADY_EXIST);
+    }
+    if (!upload) {
+      throw new Error(IMAGE_NOT_PROVIDED);
+    }
+    const uploadResult = await uploadService.uploadFiles([upload]);
```

Comment:
Why do you use `uploadFiles` for one photo if we have `uploadFile` for it?

Reply:
Replaced with string